web/src/js/components/Settings/Account/EnterEmailForm.js

```diff
@@ -0,0 +1,175 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Button from '@material-ui/core/Button'
+import Paper from '@material-ui/core/Paper'
+import { updateUserEmail } from 'js/authentication/user'
+import {
+  accountURL,
+  constructUrl,
+  loginURL,
+  replaceUrl,
+} from 'js/navigation/navigation'
+import EmailField from 'js/components/General/EmailField'
+import Typography from '@material-ui/core/Typography'
+
+export const PaperItem = props => (
+  <Paper
+    elevation={1}
+    style={{
+      padding: 24,
+      backgroundColor: '#FFF',
+    }}
+  >
+    <span
+      style={{
+        fontSize: 20,
+        fontWeight: 500,
+      }}
+    >
+      {props.title}
+    </span>
+    <Typography variant={'body2'} style={{ paddingTop: 24, paddingBottom: 24 }}>
+      {props.text}
+    </Typography>
+    <span
+      style={{
+        display: 'flex',
+        justifyContent: 'flex-end',
+        marginTop: 4,
+      }}
+    >
+      {' '}
+      {props.buttonText ? (
+        <Button
+          data-test-id={'enter-email-form-button'}
+          color={'primary'}
+          variant={'contained'}
+          onClick={props.buttonHandler}
+          style={{ minWidth: 96 }}
+        >
+          {props.buttonText}
+        </Button>
+      ) : null}
+    </span>
+  </Paper>
+)
+
+class EnterEmailForm extends React.Component {
+  constructor(props) {
+    super(props)
+    this.emailFieldRef = null
+    this.state = {
+      emailDuplicate: false,
+      emailInvalid: false,
+      otherError: false,
+      savingEmailInProgress: false,
+      verifyEmailSent: false,
+      inputEmail: null,
+    }
+  }
+
+  verifyEmailSent() {
+    this.setState({ verifyEmailSent: true })
+  }
+
+  updateUserFailure(error) {
+    if (error.code === 'auth/requires-recent-login') {
+      replaceUrl(loginURL, {
+        next: constructUrl(accountURL, { reauthed: true }),
+        reauth: 'true',
+      })
+    } else if (error.code === 'auth/email-already-in-use') {
+      this.setState({ emailDuplicate: true })
+    } else if (error.code === 'auth/internal-error') {
+      const errorMessage = JSON.parse(error.message)
+      if (errorMessage.error.message === 'INVALID_NEW_EMAIL')
+        this.setState({ emailInvalid: true })
+    }
+  }
+
+  submit() {
+    const email = this.emailFieldRef.getValue()
```

Comment:
this paradigm works, there's another way to do this that also has some benefits 

On the email input you can add a debounced onChange function that sets the email input value to state
The benefits of reading input from state are:
- you can read from state to validate
- you can make the 'submit' button disabled until it's' a valid email
- you stay in the synthetic react dom which is encouraged
I think this article takes a stronger stance than I have personally but it talks about some of the drawbacks of refs https://blog.logrocket.com/why-you-should-use-refs-sparingly-in-production/

if you do switch over to updating and reading from state you'll definitely want to debounce so you don't trigger a rerender every single time the user changes the input value